### ISSUE TITLE
array type: implement

### ISSUE THREAD

**Issue by jlapeyre** [Qiskit/openqasm3_parser#155]
- [x] oq3_lexer
- [x] oq3_parser
- [ ] oq3_syntax ? This is probably already done. Not sure
- [x] types.rs
- [ ] AST -> ASG

Notes:
* After #156. There may be some corner cases, or slightly more than corner cases that are not handled properly in the AST. But, for example array types in declaration statements are handled correctly through the AST. But not yet in the ASG.
